src/mocks/msw/handlers/tableQueryHandlers.ts

```diff
@@ -9,16 +9,51 @@ import {
   BackendDestinationEnum,
   getEndpoint,
 } from '../../../lib/utils/functions/getEndpoint'
+import {
+  BUNDLE_MASK_LAST_UPDATED_ON,
+  BUNDLE_MASK_QUERY_COLUMN_MODELS,
+  BUNDLE_MASK_QUERY_COUNT,
+  BUNDLE_MASK_QUERY_FACETS,
+  BUNDLE_MASK_QUERY_MAX_ROWS_PER_PAGE,
+  BUNDLE_MASK_QUERY_RESULTS,
+  BUNDLE_MASK_QUERY_SELECT_COLUMNS,
+  BUNDLE_MASK_SUM_FILES_SIZE_BYTES,
+} from '../../../lib/utils/SynapseConstants'
 import {
   AsynchronousJobStatus,
   AsyncJobId,
   QueryBundleRequest,
   QueryResultBundle,
 } from '../../../lib/utils/synapseTypes'
 
+const BIT_TO_FIELD_MAP: Record<number, keyof QueryResultBundle> = {
+  [BUNDLE_MASK_QUERY_RESULTS]: 'queryResult',
+  [BUNDLE_MASK_QUERY_COUNT]: 'queryCount',
+  [BUNDLE_MASK_QUERY_SELECT_COLUMNS]: 'selectColumns',
+  [BUNDLE_MASK_QUERY_MAX_ROWS_PER_PAGE]: 'maxRowsPerPage',
+  [BUNDLE_MASK_QUERY_COLUMN_MODELS]: 'columnModels',
+  [BUNDLE_MASK_QUERY_FACETS]: 'facets',
+  [BUNDLE_MASK_SUM_FILES_SIZE_BYTES]: 'sumFileSizes',
+  [BUNDLE_MASK_LAST_UPDATED_ON]: 'lastUpdatedOn',
+}
+
+function removeBundleFieldsUsingMask(
+  queryResultBundle: QueryResultBundle,
+  partMask: number,
+): QueryResultBundle {
+  const result = cloneDeep(queryResultBundle)
+  Object.entries(BIT_TO_FIELD_MAP).forEach(([bit, field]) => {
+    if ((partMask & parseInt(bit)) === 0) {
+      delete result[field]
```

Comment:
Are you cleaning up the response because the backend returns an object that contains fields that were not requested?

Reply:
Correct. It's easier to build the entire response and have MSW strip it out based on request, than to set it up for each test.

One place it's important for the response data to match this behavior is when testing pagination in QueryWrapper. We request the rows separately from the statistics, so when a user navigates to a different page, only new rows are fetched. If the response with new row data also contains non-row data, the naive merge that we do to combine the responses will overwrite the old statistics data, which wouldn't happen if that data was never returned.